src/components/application_manager/include/application_manager/application_manager_impl.h

```diff
@@ -1235,6 +1235,17 @@ class ApplicationManagerImpl
    */
   bool IsLowVoltage();
 
+  /**
+   * @brief Selects vehicle data of specified application for unsubscribing
+   * according to subscriptions of other registered applications that also
+   * subscribed to vehicle data .
+   * @param app - pointer to application instance.
+   * @return vehicle_data_for_unscribe - collection with selected vehicle data
+   * to unsubscribe.
+   */
+  application_manager::VehicleInfoSubscriptions SelectVehicleDataForUnsubscribe(
+      const app_mngr::ApplicationSharedPtr& application);
```

Comment:
@IGapchuk any reasons to pass shared pointer by reference? Also please take a look at typedef `ApplicationConstSharedPtr`

Reply:
@AKalinich-Luxoft  Fix in 11b04298dca9cf017301e7cf54e3b206cd43d883.